### DISCUSSION TITLE
Max TX Power NimBLE::getPower() vs esp_ble_tx_power_get()

### DISCUSSION THREAD

**Discussion by thekurtovic** [h2zero/NimBLE-Arduino#660] [Q&A] [answered]
Hello,
Does NimBLE arbitrarily use +9 dBm (ESP_PWR_LVL_P9) as the maximum TX power, or is this based on some specifications from Espressif?
I'm working with an ESP32-S3, and I was trying to pass ESP_PWR_LVL_P21 when specifying the TX power. It appears to work (i.e. ESP_OK) as I don't get the error message from [line 376 in NimBLEDevice.cpp](https://github.com/h2zero/NimBLE-Arduino/blob/release/1.4/src/NimBLEDevice.cpp#L374-L377).

The confusion is coming from getPower() since [the switch does not define any cases beyond +9 dBm](https://github.com/h2zero/NimBLE-Arduino/blob/release/1.4/src/NimBLEDevice.cpp#L401-L422).

If call esp_ble_tx_power_get() directly I can see that reported value coincides with the ESP_PWR_LVL_P21 enum value.
The datasheet for ESP32-S3 states that the RF power control range has a maximum 20 dBm.
Am I misunderstanding something, why doesn't NimBLE handle cases above ESP_PWR_LVL_P9?
Thanks

**Comment by h2zero** [comment, marked as answer]
This is due to not updating the max for each different chip, just a matter of code change which will be addressed soon.
